notebooks/BIA_SCRAPER.ipynb

```diff
@@ -642,446 +642,460 @@
+        "                    vterm = similar_vterm(w.lemma_.lower(), 0.86)\n",
+        "                    if vterm and 'statute' not in token.sent.text:\n",
+        "                        based_v['gender-based'] += [token.lemma_.lower()]\n",
+        "            \n",
+        "            elif similar(token.text.lower(), 'gang', 0.9):\n",
+        "                sent = token.sent\n",
+        "                based_v['gang-based'] += [token.lemma_.lower()]\n",
+        "\n",
+        "        if based_v:\n",
+        "            based_v: Dict[str, List[str]]\n",
+        "            based_v = {k:list(set(v)) for k, v in based_v.items()}\n",
+        "\n",
+        "        return based_v if based_v else None\n",
+        "\n",
+        "    def references_AB27_216(self) -> bool:\n",
+        "        '''\n",
+        "        • Returns True if the case file mentions \n",
+        "        Matter of AB, 27 I&N Dec. 316 (A.G. 2018)\n",
+        "        '''\n",
+        "        for token in self.doc:\n",
+        "            if token.text == 'I&N':\n",
+        "                sent = token.sent.text\n",
+        "                if '316' in sent and '27' in sent:\n",
+        "                    return True\n",
+        "        return False\n",
+        "\n",
+        "    def references_LEA27_581(self) -> bool:\n",
+        "        '''\n",
+        "        • Returns True if the case file mentions \n",
+        "        Matter of L-E-A-, 27 I&N Dec. 581 (A.G. 2019)\n",
+        "        '''\n",
+        "        for sent in self.doc.sents:\n",
+        "            if 'L-E-A-' in sent.text:\n",
+        "                if '27' in sent.text:\n",
+        "                    return True\n",
+        "        return False\n",
+        "                \n",
+        "    def get_seeker_sex(self) -> str:\n",
+        "        '''\n",
+        "        • This field needs to be validated. Currently, it assumes the \n",
+        "        sex of the seeker by the number of instances of pronouns in the \n",
+        "        document.\n",
+        "        '''\n",
+        "        male: int\n",
+        "        male = 0\n",
+        "\n",
+        "        female: int\n",
+        "        female = 0\n",
+        "\n",
+        "        for token in self.doc:\n",
+        "            if similar(token.text, 'he', 1) \\\n",
+        "                or similar(token.text, 'him', 1) \\\n",
+        "                or similar(token.text, 'his', 1):\n",
+        "                male += 1\n",
+        "            elif similar(token.text, 'she', 1) \\\n",
+        "                or similar(token.text, 'her', 1):\n",
+        "                female += 1\n",
+        "\n",
+        "        return 'male' if male > female \\\n",
+        "                else 'female' if female > male \\\n",
+        "                else 'unknown'\n",
+        "    \n",
+        "    def get_client_credibility(self) -> bool:\n",
```

Comment:
Looks good but in the actual class the function will need a way to get the tokens. Maybe require the string of the case file as a parameter and convert that into tokens before checking to return True or False.